src/components/TitleTransferPanel/TitleTransferPanelContent/TitleTransferPanelContent.tsx

```diff
@@ -1,23 +1,34 @@
-import React from "react";
-import { useSelector } from "react-redux";
+import React, { useState } from "react";
 import { TitleView } from "../TitleView";
 import { BLinfo } from "../TitleView/BLinfo";
+import { getTokenOwner } from "./../../../services/token";
+import { useEscrowContractUsers } from "../../../hooks";
+import { WrappedDocument } from "@govtechsg/open-attestation";
 
-export const TitleTransferPanelContent = () => {
-  const { holderAddress, beneficiaryAddress } = useSelector((state: any) => ({
-    holderAddress: state.token.holderAddress,
-    beneficiaryAddress: state.token.beneficiaryAddress
-  }));
+interface TitleTransferPanelContentProps {
+  document: WrappedDocument;
+}
+
+export const TitleTransferPanelContent = ({ document }: TitleTransferPanelContentProps) => {
+  const [tokenOwnerAddress, setTokenOwnerAddress] = useState("");
+  const { holderAddress, beneficiaryAddress } = useEscrowContractUsers({ escrowContractAddress: tokenOwnerAddress });
+  const fetchTokenOwner = async () => {
+    try {
```

Comment:
do not even try LOL

Reply:
just let it throw if you are not handling, dont do anything and the error will bubble up